src/codegen/function.rs

```diff
@@ -23,9 +25,92 @@ use std::{
     result::Result as StdResult,
 };
 
+pub fn get_must_use_if_needed(
+    parent_type_id: Option<TypeId>,
+    analysis: &analysis::functions::Info,
+    in_trait: bool,
+    only_declaration: bool,
+    comment_prefix: &str,
+) -> Option<String> {
+    if analysis.ret.bool_return_is_error.is_some()
+        || analysis.ret.nullable_return_is_error.is_some()
+    {
+        return None;
+    }
+
+    let return_type = if !analysis.outs.is_empty() {
+        let num_out_args = analysis
+            .outs
+            .iter()
+            .filter(|out| out.lib_par.array_length.is_none())
+            .count();
+        let num_out_sizes = analysis
+            .outs
+            .iter()
+            .filter(|out| out.lib_par.array_length.is_some())
+            .count();
+        // We need to differentiate between array(s)'s size arguments and normal ones. If we have 2
+        // "normal" arguments and one "size" argument, we still need to wrap them into "()" so we take
+        // that into account. If the opposite, it means that there are two arguments in any case so
+        // we need "()" too.
+        let num_outs = std::cmp::max(num_out_args, num_out_sizes);
+        match analysis.outs.mode {
+            Mode::Normal | Mode::Combined if num_outs == 1 => {
+                let array_lengths: Vec<_> = analysis
+                    .outs
+                    .iter()
+                    .filter_map(|out| out.lib_par.array_length)
+                    .collect();
```

Comment:
clippy doesn't like this `collect()`

Reply:
Ah right, need to ignore it here as well.